components/script/dom/webglprogram.rs

```diff
@@ -250,6 +293,25 @@ impl WebGLProgram {
         Ok(receiver.recv().unwrap())
     }
 
+    /// glGetProgramInfoLog
+    pub fn get_info_log(&self) -> WebGLResult<String> {
+        if self.is_deleted() {
+            return Err(WebGLError::InvalidOperation);
+        }
+        if self.link_called.get() {
+            let shaders_compiled = match (self.fragment_shader.get(), self.vertex_shader.get()) {
+                (Some(fs), Some(vs)) => fs.successfully_compiled() && vs.successfully_compiled(),
+                _ => false
+            };
+            if !shaders_compiled {
+                return Ok("One or more shaders failed to compile".to_string());
```

Comment:
Hm... Can you specify a bit more what's the error you're trying to avoid with this? `glGetProgramInfoLog` shouldn't generate an error in this case.